tests/func/test_data_cloud.py

```diff
@@ -486,7 +486,7 @@ def test_pull_partial(tmp_dir, dvc, local_remote):
     clean(["foo"], dvc)
 
     stats = dvc.pull(os.path.join("foo", "bar"))
-    assert stats["fetched"] == 2
+    assert stats["fetched"] == 3
```

Comment:
Here and below this is reverting things back to how they were https://github.com/iterative/dvc/pull/9807/files

There is an inconsistency between when directory objects are accounted for and when not, need to take another look at that specifically.